graph/floydwarshall_test.go

```diff
@@ -0,0 +1,109 @@
+package graph
+
+import (
+	"math"
+	"testing"
+)
+
+//noConnection is a value (INF) which represents that there are no two edges that directly connects these two nodes
+var noConnection = math.Inf(1)
+
+func compareDist(a, b float64) bool {
+	const float64EqualityThreshold = 1e-9
+	return math.Abs(a-b) <= float64EqualityThreshold
+}
+
+func equalMatrices(a, b [][]float64) bool {
```

Comment:
I think this should be turned into a method instead - this would ensure two things:
1. hard to misuse
2. easier to read and understand. 

My suggestion is to create a separate type

```Go
type WeightedGraph [][]float64
```

Then change the Floyd Warshall algorithm accordingly and finally create the method like so

```Go
func (w *WeightedGraph) IsAlmostEqualTo(b WeightedGraph) bool {
    ...
}
```

Reply:
Thanks!! Made this type and added this method. Can you please take a look? :)